src/main/java/com/microsoft/sqlserver/jdbc/IOBuffer.java

```diff
@@ -4664,7 +4664,8 @@ void writeTVPRows(TVP value) throws SQLServerException {
                         case VARCHAR:
                         case NCHAR:
                         case NVARCHAR:
-                            isShortValue = (2 * columnPair.getValue().precision) <= DataTypes.SHORT_VARTYPE_MAX_BYTES;
```

Comment:
columnPair.getValue().precision was 2147483647 (for varchar(max)) and the multiplication was resulting as 0.
Therefore, seeing it as a short value.